File: java/src/main/java/io/hron/Span.java
package io.hron;

/**
 * Represents a range of character positions in the input.
 *
 * @param start the start position (inclusive)
 * @param end the end position (exclusive)
 */
public record Span(int start, int end) {
  /**
   * Returns the length of this span.
   *
   * @return the number of characters covered by this span
   */
  public int length() {
    return Math.max(1, end - start);
  }
}

File: csharp/Hron/Span.cs
namespace Hron;

/// <summary>
/// Represents a range of character positions in the input.
/// </summary>
/// <param name="Start">The start position (inclusive)</param>
/// <param name="End">The end position (exclusive)</param>
public readonly record struct Span(int Start, int End)
{
    /// <summary>
    /// Returns the length of this span.
    /// </summary>
    public int Length => Math.Max(1, End - Start);
}
